impeller/compiler/types.h

```diff
@@ -16,6 +16,8 @@
 namespace impeller {
 namespace compiler {
 
+const std::string kExternalTexturePrefix = "SAMPLER_EXTERNAL_";
```

Comment:
nit:

- This should be constexpr
- We call it SAMPLER_EXTERNAL_ here but in TextureType it is ExternalOES -- would be nice if they shared the same name "SAMPLER_EXTERNAL_OES"
- Does this constant belong in types.h It's not a type, maybe it should be format.h right near the definition of TextureType::kTextureExternalOES so folks reading the code know these two concepts are tightly coupled.

Reply:
* Switched to constexpr char*.
* Done.
* Moved it to a new `constants.h`; I don't think impeller's format types are used by the compiler right now.

Reply:
(Although we could probably pull some of the format stuff from `core` into `base` at some point)